### PR TITLE
New Feature: "When to export" selector for auto-sync for NetSuite

### PR DESCRIPTION
**Pull request by twilight2294**
### Explanation of Change

Adding a new feature "When to export" selector for auto-sync for NetSuite
### Fixed Issues

$ https://github.com/Expensify/App/issues/51512
PROPOSAL: https://github.com/Expensify/App/issues/51512#issuecomment-2440145435

### Tests

Precondition: Workspace is connected to Netsuite Accounting method.

1. Go to workspace > accounting
2. Click on advanced
3. Click on auto-sync
4. Enable toggle if disabled
5. Observe that new field `when to export` appears.
6. Click on the option
7. On the new selector page, select new option

Verify that: you are navigated back to the previous page

8. Press back to go to advanced page.

Verify that: if we select `CASH` as option then the hint text below auto-sync option should be: `Out-of-pocket expenses will export when paid` and if we select `ACCURAL` as option then the hind text should be: `Out-of-pocket expenses will export when final approved`.

- [x] Verify that no errors appear in the JS console

### Offline tests

Precondition: Workspace is connected to Netsuite Accounting method.

1. Go to workspace > accounting
2. Click on advanced
3. Click on auto-sync
4. Enable toggle if disabled
5. Observe that new field `when to export` appears.
6. Click on the option
7. On the new selector page, select new option

Verify that: you are navigated back to the previous page

8. Press back to go to advanced page.

Verify that: if we select `CASH` as option then the hint text below auto-sync option should be: `Out-of-pocket expenses will export when paid` and if we select `ACCURAL` as option then the hind text should be: `Out-of-pocket expenses will export when final approved`.
### QA Steps

Precondition: Workspace is connected to Netsuite Accounting method.

1. Go to workspace > accounting
2. Click on advanced
3. Click on auto-sync
4. Enable toggle if disabled
5. Observe that new field `when to export` appears.
6. Click on the option
7. On the new selector page, select new option

Verify that: you are navigated back to the previous page

8. Press back to go to advanced page.

Verify that: if we select `CASH` as option then the hint text below auto-sync option should be: `Out-of-pocket expenses will export when paid` and if we select `ACCURAL` as option then the hind text should be: `Out-of-pocket expenses will export when final approved`.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/fa955ce7-6035-49d8-861c-bb6355f8d6dc

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/398b0fc1-cb13-4dcc-b2b2-747b1b8b2c7d

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/1df99235-0314-4624-97cd-7fc6568d3501

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/040be293-901f-4703-be97-58759952ac90

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/3d0f1b4e-f704-4e30-8abc-de876624a894

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/db4d54aa-e094-4209-8349-8e5838060b67

</details>

### CODE REVIEW

**Comment by rojiphil** [comment]
@twilight2294 Why is there a need to force-push here?
Step 15 of the guidelines [here](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#submit-your-pull-request-for-final-review) discourages us to force-push.

**Comment by rojiphil** [comment]
There are TS and Lint errors also. Please address them. 
Also, how close are we for the PR to be made ready for review?

**Comment by rojiphil** [comment]
@twilight2294 I will review it in some time. 
Meanwhile, can you please merge with the latest main? Thanks

**Review by rojiphil — COMMENTED** [review]
@twilight2294 Thanks for the PR. I have left some review comments. Please have a look.

File: src/pages/workspace/accounting/netsuite/advanced/NetSuiteAutoSyncPage.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    switchAccessibilityLabel={translate('workspace.netsuite.advancedConfig.autoSyncDescription')}
+                    shouldPlaceSubtitleBelowSwitch
+                    onCloseError={() => Policy.clearNetSuiteAutoSyncErrorField(policyID)}
+                    onToggle={(isEnabled) => Connections.updateNetSuiteAutoSync(policyID, isEnabled)}
+                    pendingAction={settingsPendingAction([CONST.NETSUITE_CONFIG.AUTO_SYNC], autoSyncConfig?.pendingFields)}
+                    errors={ErrorUtils.getLatestErrorField(autoSyncConfig, CONST.NETSUITE_CONFIG.AUTO_SYNC)}
+                />
+                {!!autoSyncConfig?.autoSync.enabled && (
+                    <MenuItemWithTopDescription
+                        title={accountingMethod === CONST.NETSUITE_ACCOUNTING_METHODS.ACCRUAL ? 'Accural' : 'Cash'}
```
[the remark this answers is not in the document]
Comment (rojiphil) [reply src/pages/workspace/accounting/netsuite/advanced/NetSuiteAutoSyncPage.tsx:56]: These should be translated depending on the `language` of the user using `advancedConfig.accountingMethods.values`

File: src/CONST.ts [hunk 2]
```diff
@@ -1863,6 +1864,11 @@ const CONST = {
         NETSUITE_ADD_CUSTOM_SEGMENT_STEP_NAMES: ['1', '2,', '3', '4', '5', '6,'],
     },
 
+    NETSUITE_ACCOUNTING_METHODS: {
+        ACCRUAL: 'ACCRUAL',
+        CASH: 'CASH',
+    },
```
[the remark this answers is not in the document]
Comment (rojiphil) [reply src/CONST.ts:1870]: As it is already defined in `expensify-common`, let us use it as we should only have a single source of truth.

**Comment by rojiphil** [comment]
> > when undefined could also mean Cash in the BE?
> 
> Is it the case ? if so i don't have any problem with our PR

We have posted [this comment](https://github.com/Expensify/App/pull/51949#issuecomment-2459467996)  to confirm the same and also to confirm the dependency between enable/disable of `autosync` and  the configured `Accounting Method`.  
Let us wait for the comments from @yuwenmemon and @dannymcclain

**Comment by dannymcclain** [comment]
That is all a bit over my head—let's wait for Yuwen to chime in.

File: src/pages/workspace/accounting/netsuite/advanced/NetSuiteAccountingMethodPage.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+type MenuListItem = ListItem & {
+    value: ValueOf<typeof CONST.NETSUITE_ACCOUNTING_METHODS>;
+};
+
+function NetSuiteAccountingMethodPage({policy}: WithPolicyConnectionsProps) {
+    const {translate} = useLocalize();
+    const policyID = policy?.id ?? '-1';
+    const styles = useThemeStyles();
+    const config = policy?.connections?.netsuite.options.config;
```
[the remark this answers is not in the document]
Comment (rojiphil) [reply src/pages/workspace/accounting/netsuite/advanced/NetSuiteAccountingMethodPage.tsx:28]: Why did we not use optional chaining here?

**Review by rojiphil — COMMENTED** [review]
@twilight2294 The changes look good. I have just one more comment left. Please have a look. Thanks.

**Comment by yuwenmemon** [comment]
> @yuwenmemon @dannymcclain Is the above understanding correct?

Explained here: https://github.com/Expensify/App/issues/51512#issuecomment-2463041361

**Comment by rojiphil** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/d12da357-f30a-4e6b-b001-2e323d7e8ab5

</details>

<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/2223c5ab-b8de-4f80-a57c-ed716b810a2c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/df97f572-009b-48f4-9870-eeaa237184ee

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/c809e586-4a94-4b0f-956f-f6d5fbe67569

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/a5e2b712-2c6c-42ac-84c2-fbd3b023e42a

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/c013ae4a-6821-4c55-a9c9-a678914129d7

</details>

**Review by rojiphil — APPROVED** [review]
Thanks @twilight2294 for all the changes.

@yuwenmemon Changes LGTM and works well too. 
Over to you for review. Thanks.

**Comment by rojiphil** [comment]
@twilight2294 Just found out an issue during offline testing as demonstrated in the test video below. Here, the auto sync status i.e. `Enabled` and `Disabled` should be grayed when changed. Similarly, on changing `Cash` and `Accrual` accounting method, the changes should be grayed. Please fix this too. Thanks.

@yuwenmemon Please hold review until this is integrated too. 

https://github.com/user-attachments/assets/46637059-62b9-4811-9f7e-8b22c41cba54

**Comment by rojiphil** [comment]
> what should be the behaviour on `Advanced` page when someone changes the accounting method? the hint text changes so should we gray out the content in this case to? the current behaviour is as follows:

When the `accounting method` is changed, I think the greying out should propagate even to `Auto sync` and the parent `Advanced`. A similar pattern is seen in `Workflows->Delay Submissions`  as demonstrated in video below i.e. when `Submission frequency` is changed, the greying propagates even to `Delay Submissions`. 
Would this not address the greying of the hint text as well? 


https://github.com/user-attachments/assets/94d104f6-acac-49e4-8b48-a3cf39221bd9

**Review by rojiphil — APPROVED** [review]
Thanks @twilight2294 for the changes. 

@yuwenmemon LGTM and works well in offline too. 
All yours for review. Thanks


https://github.com/user-attachments/assets/470f0985-970b-49d8-943f-1e9b756228c6

File: src/pages/workspace/accounting/netsuite/advanced/NetSuiteAccountingMethodPage.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            </View>
+        ),
+        [translate, styles.pb5, styles.ph5],
+    );
+
+    const selectExpenseReportApprovalLevel = useCallback(
+        (row: MenuListItem) => {
+            if (row.value !== config?.accountingMethod) {
+                Connections.updateNetSuiteAccountingMethod(policyID, row.value, config?.accountingMethod ?? COMMON_CONST.INTEGRATIONS.ACCOUNTING_METHOD.CASH);
+            }
```
Comment (yuwenmemon) [inline src/pages/workspace/accounting/netsuite/advanced/NetSuiteAccountingMethodPage.tsx:56]: NAB: Add a comment here explaining why we're defaulting to CASH

**Review by yuwenmemon — APPROVED** [review]

**Comment by yuwenmemon** [comment]
Tested as well, works great! Thanks!

**Pull request merged**